### PR TITLE
accelerators: (snax-gemmx) add initial accelerator description

### PR DESCRIPTION
**Pull request by jorendumoulin**

### CODE REVIEW

File: compiler/accelerators/snax_gemmx.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    ) -> Sequence[Operation]:
+        """
+        Lowers the operation op to a sequence of acc_ops.
+        acc_ops are:
+            - *.op that generates SSAValues consumed by acc2.setup
+            - acc2.setup
+            - acc2.launch
+            - acc2.await
+        These ops can further be lowered by specific instances of the
+        Accelerator interface
```
Comment (JosseVanDelm) [inline compiler/accelerators/snax_gemmx.py:122]: Maybe remove / adjust this comment specific to this accelerator?

File: compiler/accelerators/snax_gemmx.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+            self.name,
+            {
+                **streamer_setup,
+                "K": addr_next + 0,
+                "N": addr_next + 1,
+                "M": addr_next + 2,
+                "subtractions": addr_next + 3,
+                "csr0": addr_next + 4,
+                "csr1": addr_next + 5,
+                "csr2": addr_next + 6,
```
Comment (JosseVanDelm) [inline compiler/accelerators/snax_gemmx.py:101]: what are these for?

File: compiler/accelerators/snax_gemmx.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    ) -> Sequence[tuple[Sequence[Operation], SSAValue]]:
+        """
+        Produce a `Sequence[Operation], SSAValue` tuple
+        for each field that contains:
+
+        - a list of operations that calculate the field value
+        - a reference to the SSAValue containing the calculated field value
+        """
+
+        raise NotImplementedError()
```
Comment (JosseVanDelm) [inline compiler/accelerators/snax_gemmx.py:152]: Is this for  a next PR?

File: compiler/accelerators/snax_gemmx.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+    @staticmethod
+    def lower_acc_await(acc_op: accfg.AcceleratorOp) -> Sequence[Operation]:
```
Comment (JosseVanDelm) [inline compiler/accelerators/snax_gemmx.py:155]: Any reason why we are not putting this with the other barrier mechanisms?

File: compiler/accelerators/snax_gemmx.py (hunk 3 above)
Comment (jorendumoulin) [reply compiler/accelerators/snax_gemmx.py:152]: yep

File: compiler/accelerators/snax_gemmx.py (hunk 2 above)
Comment (jorendumoulin) [reply compiler/accelerators/snax_gemmx.py:101]: https://github.com/KULeuven-MICAS/snax_cluster/blob/a182ddc0c50ef7b64d14c7c8222dcba0cb009f14/target/snitch_cluster/sw/snax/gemmx/src/snax-gemmx-lib.c#L23
bin packed csr registers to set a variety of values. although the naming is very bad i kept it to stay consistent with the naming in hw. I"ll add some comments explaining them

File: compiler/accelerators/snax_gemmx.py (hunk 4 above)
Comment (jorendumoulin) [reply compiler/accelerators/snax_gemmx.py]: i moved it over there, welcome `SNAXPollingBarrier4`!

File: compiler/accelerators/snax_gemmx.py (hunk 2 above)
Comment (JosseVanDelm) [reply compiler/accelerators/snax_gemmx.py:111]: :+1: :frowning_face: :roll_eyes:  :upside_down_face: :+1:

**Review by JosseVanDelm — APPROVED** [review]
Lgtm! Thank you Joren!

**Pull request merged**